kratos/processes/check_same_modelpart_using_skin_distance_process.cpp

```diff
@@ -0,0 +1,179 @@
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Vicente Mataix Ferrandiz
+//
+//
+
+// System includes
+#include <limits>
+
+// External includes
+
+// Project includes
+#include "geometries/hexahedra_3d_8.h"
+#include "processes/check_same_modelpart_using_skin_distance_process.h"
+#include "utilities/auxiliar_model_part_utilities.h"
+// #include "processes/calculate_distance_to_skin_process.h" // Continuous version
+#include "processes/calculate_discontinuous_distance_to_skin_process.h" // Discontinuous version
+#include "processes/structured_mesh_generator_process.h"
+
+namespace Kratos
+{
+template<std::size_t TDim>
+void CheckSameModelPartUsingSkinDistanceProcess<TDim>::Execute()
+{
+    KRATOS_TRY
+
+    // We get the names of the model parts
+    // const std::string& r_model_part_1_name = mThisParameters["model_part_1_name"].GetString();
+    // const std::string& r_model_part_2_name = mThisParameters["model_part_2_name"].GetString();
+    const std::string& r_skin_model_part_1_name = mThisParameters["skin_model_part_1_name"].GetString();
+    const std::string& r_skin_model_part_2_name = mThisParameters["skin_model_part_2_name"].GetString();
+
+    // We get the model parts
+    // ModelPart& r_model_part_1 = mrModel.GetModelPart(r_model_part_1_name);
+    ModelPart& r_skin_model_part_1 = mrModel.GetModelPart(r_skin_model_part_1_name);
+    // ModelPart& r_model_part_2 = mrModel.GetModelPart(r_model_part_2_name);
+    ModelPart& r_skin_model_part_2 = mrModel.GetModelPart(r_skin_model_part_2_name);
+
+    // We get the coordinates of the bounding box
+    using NodeType = Node<3>;
+    const double bounding_box_scale_factor = mThisParameters["bounding_box_scale_factor"].GetDouble();
+    const double max_x = bounding_box_scale_factor * block_for_each<MaxReduction<double>>(r_skin_model_part_1.Nodes(), [&](NodeType& rNode) {
+        return rNode.X();
+    });
+    const double min_x = bounding_box_scale_factor * block_for_each<MinReduction<double>>(r_skin_model_part_1.Nodes(), [&](NodeType& rNode) {
+        return rNode.X();
+    });
+    const double max_y = bounding_box_scale_factor * block_for_each<MaxReduction<double>>(r_skin_model_part_1.Nodes(), [&](NodeType& rNode) {
+        return rNode.Y();
+    });
+    const double min_y = bounding_box_scale_factor * block_for_each<MinReduction<double>>(r_skin_model_part_1.Nodes(), [&](NodeType& rNode) {
+        return rNode.Y();
+    });
+    const double max_z = bounding_box_scale_factor * block_for_each<MaxReduction<double>>(r_skin_model_part_1.Nodes(), [&](NodeType& rNode) {
+        return rNode.Z();
+    });
+    const double min_z = bounding_box_scale_factor * block_for_each<MinReduction<double>>(r_skin_model_part_1.Nodes(), [&](NodeType& rNode) {
+        return rNode.Z();
```

Comment:
use a combined reduction, then you only need to loop once over the nodes

Also why not use the existing BB?

Reply:
First thing, true. Second thing, the structured mesher expects a hexahedra or quadrilateral